File: tests/machine/x/lua/short_circuit.lua
function __print(...)
    local args = {...}
    local parts = {}
    for i,a in ipairs(args) do
        if a ~= nil and a ~= '' then parts[#parts+1] = tostring(a) end
    end
    print(table.concat(parts, ' '))
end
function boom(a, b)
  __print("boom")
  return true
end

__print((false and boom(1, 2)))
__print((true or boom(1, 2)))

File: tests/machine/x/cpp/short_circuit.cpp
#include <iostream>
#include <string>

bool boom(int a, int b) {
  std::cout << std::string("boom") << std::endl;
  return true;
}

int main() {
  std::cout << ((false && boom(1, 2)) ? "true" : "false") << std::endl;
  std::cout << ((true || boom(1, 2)) ? "true" : "false") << std::endl;
  return 0;
}
